sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -135,6 +135,24 @@ abstract class JdbcDialect extends Serializable with Logging {
     s""""$colName""""
   }
 
+  /**
+   * Get the SQL that should be used to create a table. Dialects can
+   * override this method to return a SQL that works best in a particular database.
+   * To allow certain options to append when create a new table, which can be
+   * table_options or partition_options.
+   * E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
+   * @param tableName The name of the table.
+   * @param strSchema The scheme string corresponding to the dialect, the schemaString method of
+   *                  JdbcUtils has provided the function of calculating the scheme string
+   *                  for different databases
+   * @param createTableOptions The string of the options that to allow certain options to append
+   *                           when create a new table, which can be table_options
+   *                           or partition_options.
+   */
+  def createTable(tableName: String, strSchema: String, createTableOptions: String): String = {
```

Comment:
shall we make it consistent with `createSchema`, and let the implementation execute the create table statement instead of only generating the SQL?